src/averageAges.js

```diff
@@ -55,7 +66,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const child = people.filter(children => onlyWithSon
+    ? people.some(person => children.mother === person.name)
+      && children.sex === 'm'
+    : people.some(person => children.mother === person.name)
+  );
+
+  const averageAgeDiff = child.reduce(
+    (sum, children) => sum + (children.born - people.find(mother =>
+      children.mother === mother.name).born),
+    0)
+    / child.length;
```

Comment:
I guess, too many operations at a time, you could split it, but still – up to you
```
const averageAgeDiff = child.reduce((sum, children) => {
  const mother = people.find(mother => children.mother === mother.name);
  ...
}, 0);
```
however, let's make it more readable